### ISSUE TITLE
Proper Inventory script usage, expected JSON output format, more verbose errors?

### ISSUE THREAD

**Issue by cidrbl0ck** [ansible/ansible#80794]
### Summary

I've been working on trying to get a working Inventory script for a month now and while normally the documentation is pretty awesome.. anything I can find for the script module is all but bare.
I realize folks would rather people created plugins, however in my situation a script makes more sense and is more doable.

Anyway the page here: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/script_inventory.html#script-inventory

Really needs to be fleshed out with an example of the required JSON block that will be ingested along with how to test using ansible-inventory. I'd also like to mention that in my battle trying to get this to work, the resulting errors once might receive are super not helpful in determining what went wrong.

I don't want to segway off into this being a bug report but suffice to say I am returning valid JSON, but the controller claims otherwise. And I have no actual documentation to help troubleshoot this issue with.

One specific error condition that I hope can be improved: 
` Expecting value: line 1 column 1 (char 0).
Expecting value: line 1 column 1 (char 0)`

If there's any way that this could actually return the values (or line) in question that'd be much more helpful than the existing text.

Thanks!

### Issue Type

Documentation Report

### Component Name

script

### Ansible Version

```console
$ ansible --version
ansible [core 2.14.4]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/runner/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.9/site-packages/ansible
  ansible collection location = /home/runner/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.9.13 (main, Nov  9 2022, 13:16:24) [GCC 8.5.0 20210514 (Red Hat 8.5.0-15)] (/usr/bin/python3.9)
  jinja version = 3.1.2
  libyaml = True
```

### Configuration

```console
# if using a version older than ansible-core 2.12 you should omit the '-t all'
$ ansible-config dump --only-changed -t all
bash-4.4# ansible-config dump --only-changed -t all
CONFIG_FILE() = /etc/ansible/ansible.cfg
```

### OS / Environment

ansible23/ee-supported-rhel8

### Additional Information

Having asked around online in various places, I am not the only one who relies upon this module and also who has had trouble using it.
Now I do agree that building a plugin might be better overall, but not everyone has the skills or the access to those that do to accomplish this. So having and using the script module is our Obi Wan, please help us use it correctly.

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/modules/script.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/script.py)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/devel/ISSUE_HELP.md)

**Comment by cidrbl0ck** [comment]
Thanks Brian.. So the link above, same as the one in your PR doesn't contain any differing information than the leader page to the link I embedded. I was asking for the info to be expanded upon which, to be fair, could go into either page. But yeah, the existing publicly available inventory script/plugin page(s) desperately need more information. The 'developing inventory scripts' page is 75% about a plugin.

**Comment by cidrbl0ck** [comment]
That's all fair Brian. I opened a BR with all the relevant info pertaining to the problem I'm fighting.   I still believe the info currently on those pages is not sufficient for anyone not intimate with the underlying processes. 

Either I'm still doing it wrong, or there's a real bug.

**Comment by s-hertel** [comment]
@cidrbl0ck You haven't provided context to debug (the script you're trying to write, command you're running, and the full output), but based on `Expecting value: line 1 column 1 (char 0). Expecting value: line 1 column 1 (char 0)` I'm guessing your script isn't printing anything at all.

**Comment by cidrbl0ck** [comment]
@s-hertel So oddly enough this is now working as expected. The resolution is entirely unknown, TL;DR I had thought the problem was credential related.. since you cannot assign more than one credential in AAP to an inventory source I created a new type that was the DB creds used originally combined with the typical set used by all other templates.  The inventory ended up running perfectly, however my RH guy wanted to test further so we switched back to the OG set and it still worked perfectly. 
We spent the next 2 hours trying all manner of things to figure out why and we could not. So, while there's no idea what the actual root cause was, the scripts contents remain unchanged from when I submitted this issue. 

One thing that we did discover while testing various things; if you break the DB connection. Say fat-finger the password value in the environment variable, when AAP tries to run the Job it'll fail with an identical error as above:
 `Expecting value: line 1 column 1 (char 0). Expecting value: line 1 column 1 (char 0)`

Ansible-Inventory seems to ignore any error handling within the script.. I imagine there might be good reasons for this but it certainly does not help when testing things.

Before you jump and say that my credentials were wrong.. they most assuredly were not. My RH guy verified, and I had a co-worker validate as well.  If you typo the password it'll return the error above. If you comment out the `print(json.dumps(inventory, indent=4))` line, it'll return the same thing.  With the former, ignoring the `exception` I have for failing to connect to the DB. 

He submitted an RFE from his side regarding error handling, I'd like to ask about the same in here but I understand if my welcome has been worn out.

Here's the script: https://github.com/bicarbonate/ansible_phpipam
